# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/02 10:46:04-08:00 rusty@rustcorp.com.au 
#   [PATCH] Fix proc removal in ip_conntrack_standalone
#   
#   Someone thought it would be clever if proc code ignores removal of
#   non-existent entries.  Hence, we missed that /proc/net/stat/ip_conntrack
#   is never removed on module removal or init failure.
#   
#   Found by nfsim.
#   
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# net/ipv4/netfilter/ip_conntrack_standalone.c
#   2004/12/29 22:39:53-08:00 rusty@rustcorp.com.au +2 -2
#   Fix proc removal in ip_conntrack_standalone
# 
diff -Nru a/net/ipv4/netfilter/ip_conntrack_standalone.c b/net/ipv4/netfilter/ip_conntrack_standalone.c
--- a/net/ipv4/netfilter/ip_conntrack_standalone.c	2005-01-04 18:27:19 -08:00
+++ b/net/ipv4/netfilter/ip_conntrack_standalone.c	2005-01-04 18:27:19 -08:00
@@ -820,8 +820,8 @@
 	nf_unregister_hook(&ip_conntrack_defrag_ops);
  cleanup_proc_stat:
 #ifdef CONFIG_PROC_FS
-	proc_net_remove("ip_conntrack_stat");
-cleanup_proc_exp:
+	remove_proc_entry("ip_conntrack", proc_net_stat);
+ cleanup_proc_exp:
 	proc_net_remove("ip_conntrack_expect");
  cleanup_proc:
 	proc_net_remove("ip_conntrack");
